src/cas/abc_cas_converter.py

```diff
@@ -0,0 +1,258 @@
+
+    """
+    for row in cat.itertuples():
+        labelset = row.cluster_annotation_term_set_label
+        cell_label = row.name
+        cell_ontology_term_id = None
+        cell_ids = None
+        rationale = None
+        rationale_dois = None
+        marker_gene_evidence = None
+        synonyms = None
+        category_fullname = None
+        category_cell_ontology_exists = None
+        category_cell_ontology_term_id = None
+        category_cell_ontology_term = None
+        cell_set_accession = row.label
+        parent_cell_set_accession = row.parent_term_label
+
+        anno = {
+            "labelset": labelset,
+            "cell_label": cell_label,
+            "cell_fullname": cell_label,
+            "cell_ontology_term_id": cell_ontology_term_id,
+            "cell_ontology_term": cell_label,
+            "cell_ids": cell_ids,
+            "rationale": rationale,
+            "rationale_dois": rationale_dois,
+            "marker_gene_evidence": marker_gene_evidence,
+            "synonyms": synonyms,
+            "category_fullname": category_fullname,
+            "category_cell_ontology_exists": category_cell_ontology_exists,
+            "category_cell_ontology_term_id": category_cell_ontology_term_id,
+            "category_cell_ontology_term": category_cell_ontology_term,
+            "cell_set_accession": cell_set_accession,
+            "parent_cell_set_accession": parent_cell_set_accession,
+        }
+        cas.get("annotations").append({k: v for k, v in anno.items() if v is not None and not pd.isna(v)})
+
+
+def add_labelsets(cas: Dict[str, Any], cat_set: pd.DataFrame):
+    """
+    Adds labelsets to the Cell Annotation Schema (CAS) based on the data from the Cluster Annotation Term Set DataFrame.
+
+    Args:
+        cas (Dict[str, Any]): Cell Annotation Schema dictionary.
+        cat_set (pandas.DataFrame): DataFrame containing Cluster Annotation Term Set data.
+    """
+    order_mapping = calculate_order_mapping(cat_set["order"])
+    for row in cat_set.itertuples():
+        name = row.label
+        description = row.description
+        rank = None
+        if row.order != 0:
+            rank = order_mapping[row.order]
+
+        labelset = {"name": name, "description": description}
+        if rank is not None:
+            labelset["rank"] = rank
+        cas.get("labelsets").append(labelset)
+
+
+def init_metadata() -> Dict[str, Any]:
+    """
+    Initializes metadata for Cell Annotation Schema (CAS).
+
+    Returns:
+        Dict[str, Any]: Metadata dictionary containing default values for various fields.
+    """
+    # TODO These needs proper assignments
+    matrix_file_id = None
+    cellannotation_schema_version = None
+    cellannotation_timestamp = None
+    cellannotation_version = None
+    cellannotation_url = None
+    author_name = None
+    author_contact = None
+    orcid = None
+    cas = {
+        "matrix_file_id": matrix_file_id,
```

Comment:
Can we inline above variable declarations as well?

Reply:
I would like to keep it as it is for now, as transitioning to a CAS object would necessitate a more in-depth refactoring of our code